rustls/src/client/tls12.rs

```diff
@@ -1036,36 +951,22 @@ impl ExpectFinished {
 impl State<ClientConnectionData> for ExpectFinished {
     fn handle(self: Box<Self>, cx: &mut ClientContext<'_>, m: Message) -> hs::NextStateOrError {
         let mut st = *self;
-        let finished =
-            require_handshake_msg!(m, HandshakeType::Finished, HandshakePayload::Finished)?;
-
-        cx.common.check_aligned_handshake()?;
-
-        // Work out what verify_data we expect.
-        let vh = st.transcript.get_current_hash();
-        let expect_verify_data = st.secrets.server_verify_data(&vh);
 
-        // Constant-time verification of this is relatively unimportant: they only
-        // get one chance.  But it can't hurt.
-        let _fin_verified =
-            constant_time::verify_slices_are_equal(&expect_verify_data, &finished.0)
-                .map_err(|_| {
-                    cx.common
-                        .send_fatal_alert(AlertDescription::DecryptError);
-                    Error::DecryptError
-                })
-                .map(|_| verify::FinishedMessageVerified::assertion())?;
+        let fin_verified = tls12::hs::handle_finished(
+            &st.secrets,
+            &st.transcript,
+            cx.common,
+            tls12::SERVER_FINISHED,
+            &m,
+        )?;
```

Comment:
I was surprised this PR is currently only net -132 lines of code (+416/−548). It seems like the styling of function calls like this makes measuring in LoC quite misleading.